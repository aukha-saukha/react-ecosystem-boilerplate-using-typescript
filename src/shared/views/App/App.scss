.wrapper {
  display: grid;
  grid-template-areas:
    'header'
    'page-specific-content'
    'footer';
  grid-template-rows: auto 1fr auto;
  margin: 0 auto;
  max-width: 1440px;
  min-height: 100vh;
  min-width: 300px;
}

.page-specific-content {
  grid-area: page-specific-content;
}
