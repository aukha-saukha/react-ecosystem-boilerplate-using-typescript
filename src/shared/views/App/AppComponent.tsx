import * as React from 'react';
import { hot } from 'react-hot-loader/root';

import '../../../static/css/global.scss';
import styles from './App.scss';

const App: React.FC<{}> = (): JSX.Element => (
  <div className={`font-size-12 ${styles['greeting']}`}>Hello World!</div>
);

export default hot(App);
