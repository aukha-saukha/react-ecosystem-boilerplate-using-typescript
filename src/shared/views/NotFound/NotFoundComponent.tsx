import * as React from 'react';

const NotFoundComponent: React.FC<{}> = (): JSX.Element => (
  <main className="font-size-12">Four oh! Four</main>
);

export default NotFoundComponent;
