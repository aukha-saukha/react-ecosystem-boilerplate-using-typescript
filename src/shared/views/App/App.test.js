import { mount, shallow } from 'enzyme';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import App from './AppComponent';
import NotFound from '../NotFound';

describe('App component', () => {
  it('Invalid path renders "Not Found" page', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/random']}>
        <App />
      </MemoryRouter>
    );
    expect(wrapper.find(NotFound)).toHaveLength(1);

    wrapper.unmount();
  });

  it('Renders correctly', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
});
