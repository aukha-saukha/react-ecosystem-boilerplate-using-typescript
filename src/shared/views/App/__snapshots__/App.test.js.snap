// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Renders correctly 1`] = `
<ErrorBoundaryComponent>
  <div
    className="wrapper"
  >
    <HeaderComponent />
    <section
      className="padding-all-15 page-specific-content"
    >
      <Switch>
        // TODO: Check if any type can be avoided below.
        <Route
          exact={true}
          key="/about-us"
          path="/about-us"
          render={[Function]}
        />
        <Route
          exact={true}
          key="/contact-us"
          path="/contact-us"
          render={[Function]}
        />
        <Route
          exact={true}
          key="/"
          path="/"
          render={[Function]}
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </section>
    <FooterComponent />
  </div>
</ErrorBoundaryComponent>
`;
